colorizer_data/types.py

```diff
@@ -97,15 +100,64 @@ class FrameDimensions(TypedDict):
     """Height of a frame in physical units (not pixels)."""
 
 
-class DatasetMetadata(TypedDict):
+class BaseMetadataJson(TypedDict):
+    """JSON dictionary format for `BaseMetadata`."""
+
+    name: str
+    description: str
+    author: str
+    dateCreated: str
+    lastModified: str
+    revision: str
+    dataVersion: str
+
+
+@dataclass
+class BaseMetadata:
+    """Shared metadata between datasets and collection files."""
+
+    name: str = None
+    description: str = None
+    author: str = None
+    date_created: str = None
+    """Formatted datetime string. See `DATETIME_FORMAT`. """
+    last_modified: str = None
+    """Formatted datetime string. See `DATETIME_FORMAT`. """
+    revision: int = None
+    """
+    Revision number. Will be updated each time the dataset or collection
+    is rewritten. Starts at 0.
+    """
+    data_version: str = CURRENT_VERSION
```

Comment:
is this always incremented in lockstep or can collection version number be totally independent of dataset version number? I'm assuming we want them to be allowed to have different version numbers.

Reply:
Ooops I misread the comments. I always think the comment precedes the code.
I think data_version should be named writer_version (or [packagename]_version) and the above comment I posted applies to "revision".

Reply:
Added a new separate string parameter, `dataset_version` and `collection_version`.